sunpy/timeseries/sources/goes.py

```diff
@@ -214,14 +184,58 @@ def _parse_hdus(cls, hdulist):
                              ('xrsb', u.W/u.m**2)])
         return data, header, units
 
+    @staticmethod
+    def _parse_netcdf(filepath):
+        """
+        Parses the netCDF GOES files to return the data, header and associated units.
+
+        Parameters
+        ----------
+        filepath : `~str`
+            The path of the file to parse
+        """
+        d = h5netcdf.File(filepath, mode="r")
+
+        header =  MetaDict(OrderedDict(d.attrs))
+        if "a_flux" in d.variables:
+            xrsa = np.array(d["a_flux"])
+            xrsb = np.array(d["b_flux"])
+            start_time_str = d["time"].attrs["units"].astype(str).lstrip("seconds since").rstrip("UTC")
+            times = parse_time(start_time_str) + TimeDelta(d["time"], format="sec")
+        elif "xrsa_flux" in d.variables:
+            xrsa = np.array(d["xrsa_flux"])
+            xrsb = np.array(d["xrsb_flux"])
+            start_time_str = d["time"].attrs["units"].astype(str).lstrip("seconds since")
+            times = parse_time(start_time_str) + TimeDelta(d["time"], format="sec")
+
+        else:
+            raise ValueError(f"The file {filepath} doesn't seem to be a GOES netcdf file.")
+
+        d.close()
+
+        data = DataFrame({"xrsa": xrsa, "xrsb": xrsb}, index=times.datetime)
+        data.replace(-9999, np.nan)
+        units = OrderedDict([("xrsa", u.W/u.m**2),
+                             ("xrsb", u.W/u.m**2)])
+
+        return data, header, units
+
     @classmethod
     def is_datasource_for(cls, **kwargs):
         """
         Determines if header corresponds to a GOES lightcurve
         `~sunpy.timeseries.TimeSeries`.
         """
-        if 'source' in kwargs.keys():
-            if kwargs.get('source', ''):
-                return kwargs.get('source', '').lower().startswith(cls._source)
-        if 'meta' in kwargs.keys():
-            return kwargs['meta'].get('TELESCOP', '').startswith('GOES')
+        if "source" in kwargs.keys():
+            if kwargs.get("source", ""):
+                return kwargs.get("source", "").lower().startswith(cls._source)
+        if "meta" in kwargs.keys():
+            return kwargs["meta"].get("TELESCOP", "").startswith("GOES")
+
+        if "filepath" in kwargs.keys():
+            try:
+                if sunpy.io.detect_filetype(kwargs.get("filepath", "")) == "nc":
+                    with h5netcdf.File(kwargs.get("filepath", ""), mode="r") as f:
+                        return "XRS" in f.attrs["summary"].astype("str")
+            except:
```

Comment:
```suggestion
            except Exception as e:
```
This should placate `pre-commit`

Reply:
great thanks @dstansby !